### DISCUSSION TITLE
Using GenerateAuthToken(ctx context.Context, region string) with Franzgo giving error "failed to load credentials: failed to refresh cached credentials, failed to retrieve credentials, not found, Signing"

### DISCUSSION THREAD

**Discussion by tshinde-incontact** [aws/aws-msk-iam-sasl-signer-go#20] [Q&A]
We are using the aws-msk-iam-sasl-signer-go library with our Golang microservice deployed on EKS.

A service account role has been provided, and we validated the associated policies with the AWS support team, who confirmed that they are correct. The role includes a trust relationship with the proper sts:AssumeRoleWithWebIdentity rule.

Here is the code snippet we are using to obtain the token:

```
 `func signCredentials(opts []kgo.Opt) []kgo.Opt {
	opts = append(
		opts,
		kgo.SASL(oauth.Oauth(func(ctx context.Context) (oauth.Auth, error) {
			token, _, err := signer.GenerateAuthToken(context.TODO(), AwsRegion)
			if err != nil {
				log.Fatalf("Failed to generate auth token: %v", err)
			}
			return oauth.Auth{Token: token}, nil
		})),
	)
	return opts
}`
```

Could you please assist in identifying what might be causing the issue?

**Comment by nice-hitendrapatil** [comment]
Can someone please help with the above ask?
